datamodel/include/datamodel/rendering/BufferObject.hpp

```diff
@@ -14,11 +15,33 @@ namespace rcbe::rendering {
 
 class VertexBufferObject {
 public:
+
     using ValueType = GLfloat;
     using StorageType = std::vector<ValueType>;
     using RawDataType = const ValueType *;
 
-    explicit VertexBufferObject(const std::vector<rcbe::geometry::Mesh>& meshes);
+    class VertexArrayObject {
+    public:
+        VertexArrayObject();
+        ~VertexArrayObject();
+
+        /*VertexArrayObject(const VertexArrayObject &other) = default;
```

Comment:
You need to fix these. Basically I presume it would be better if these are move only.